src/status_im2/common/toasts/events.cljs

```diff
@@ -0,0 +1,48 @@
+(ns status-im2.common.toasts.events
+  (:require
+   [taoensso.encore :as enc]
+   [utils.re-frame  :as rf]))
+
+(def ^:private next-toast-id (atom 0))
+
+(rf/defn upsert
+  {:events [:toasts/upsert]}
+  [{:keys [db]} id opts]
+  (let [{:toasts/keys [index toasts]} db
+        update?                       (some #{id} index)
+        index                         (enc/conj-when index (and (not update?) id))
```

Comment:
can `enc/conj-when` be replaced by core functions ? the problem in using such sugar is that devs must be familiar with it, overwise it will take time to read the code, so its always better to use core functions

Reply:
done